src/BuildScriptGeneratorCli/ServiceProviderBuilder.cs

```diff
@@ -29,19 +31,38 @@ public ServiceProviderBuilder(string logFilePath = null, IConsole console = null
             LogManager.Configuration = BuildNLogConfiguration(logFilePath);
             LogManager.ReconfigExistingLoggers();
 
+            var disableTelemetryEnvVariableValue = Environment.GetEnvironmentVariable(
+               LoggingConstants.OryxDisableTelemetryEnvironmentVariableName);
+            _ = bool.TryParse(disableTelemetryEnvVariableValue, out bool disableTelemetry);
+            var config = new TelemetryConfiguration();
+            var aiKey = disableTelemetry ? null : Environment.GetEnvironmentVariable(
+                LoggingConstants.ApplicationInsightsConnectionStringKeyEnvironmentVariableName);
+            if (!string.IsNullOrWhiteSpace(aiKey))
+            {
+                config.ConnectionString = aiKey;
+            }
+
             this.serviceCollection = new ServiceCollection();
             this.serviceCollection
                 .AddBuildScriptGeneratorServices()
                 .AddCliServices(console)
                 .AddLogging(builder =>
                 {
+                    if (!string.IsNullOrWhiteSpace(aiKey))
+                    {
+                        builder.AddApplicationInsights(
```

Comment:
If `aiKey` is `null`, then we don't run `AddApplicationInsights` -- do you know if this affects the `ILogger` object we try to retrieve via dependency injection?

Reply:
Do not think this would affect the `ILogger` object. My understanding is that if the `aiKey` is null, the logs collected from ILogger would _not_ be submitted to AI because `AddApplicationInsights` would not run.
I quickly tested by assigning a null value to `aiKey` and spun up a container with this new image, was able to still successfully build the application. Any reason why you think this may affect the `ILogger` object?

Reply:
I wasn't sure if the `ILogger` that we were pulling from dependency injection was set up with the `.AddLogging` or `.AddApplicationInsights` call -- if it's set up with the latter, then my concern would be `aiKey` having a `null` or empty value and we avoid setting up the dependency injection for `ILogger` since `.AddApplicationInsights` is not called, and when we request an `ILogger` in a class constructor, there would be an exception thrown because DI doesn't know how to provide an object of that type.

If you tested it locally without providing a value for `aiKey` and no exceptions were thrown, then it seems like `ILogger` is set up in the `.AddLogging` call, which we're making no matter what, so we should be good here.